File: frontend/src/types/Product.ts
interface Product {
  id?: number;
  name: string;
  description: string;
  createdAt?: string;
  price: number;
}

export function createEmptyProduct(): Product {
  return {
    name: '',
    description: '',
    price: 0,
  };
}

export type { Product };

File: backend-java/src/main/java/com/canadel/entity/Product.java
package com.canadel.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Data;
import java.util.Date;

@Data
@Entity
public class Product {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;

  private String name;

  private String description;

  private Double price;

  private Date createdAt;
}
